tracing-error/src/backtrace.rs

```diff
@@ -0,0 +1,65 @@
+use super::try_bool;
+use crate::layer::WithContext;
+use std::fmt;
+use tracing::{Metadata, Span};
+
+#[derive(Clone)]
+pub struct SpanBacktrace {
+    span: Span,
+}
+
+// === impl SpanBacktrace ===
+
+impl SpanBacktrace {
+    pub fn capture() -> Self {
+        SpanBacktrace {
+            span: Span::current(),
+        }
+    }
+
+    pub fn with_spans(&self, f: impl FnMut(&'static Metadata<'static>, &str) -> bool) {
+        self.span.with_subscriber(|(id, s)| {
+            if let Some(getcx) = s.downcast_ref::<WithContext>() {
+                getcx.with_context(s, id, f);
+            }
+        });
+    }
+}
+
+impl fmt::Debug for SpanBacktrace {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut err = None;
+        let mut span = 0;
+
+        writeln!(f, "span backtrace:")?;
+        self.with_spans(|metadata, fields| {
+            try_bool!(
+                writeln!(f, "{:>4}: {}::{}", span, metadata.target(), metadata.name()),
```

Comment:
it would be really nice if each span in the trace could inherit the width, padding, and alignment of the root formatter, so you can say things like `format!("{:<80}", span_backtrace)` and have your formatting preferences honored...it might be tricky to make this work nicely so i am okay with punting on it...

Reply:
Not sure how this would work but I'm down to try.